its/src/test/java/com/sonar/it/csharp/MultipleProjectsTest.java

```diff
@@ -0,0 +1,422 @@
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_SECOND_CLASS_FILE, "files")).isEqualTo(1);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_ASSEMBLY_INFO_FILE, "files")).isEqualTo(1);
+
+    assertThat(getFileMeasureAsInt(SECOND_PROJECT_FIRST_CLASS_FILE, "files")).isEqualTo(1);
+  }
+
+  /* Statements */
+
+  @Test
+  public void statementsAtProjectLevel() {
+    assertThat(getProjectMeasureAsInt("statements")).isEqualTo(13);
+  }
+
+  @Test
+  public void statementsAtDirectoryLevel() {
+    assertThat(getDirectoryMeasureAsInt(FIRST_PROJECT_DIRECTORY, "statements")).isEqualTo(10);
+    assertThat(getDirectoryMeasureAsInt(SECOND_PROJECT_DIRECTORY, "statements")).isEqualTo(3);
+  }
+
+  @Test
+  public void statementsAtFileLevel() {
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_FIRST_CLASS_FILE, "statements")).isEqualTo(8);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_SECOND_CLASS_FILE, "statements")).isEqualTo(2);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_ASSEMBLY_INFO_FILE, "statements")).isEqualTo(0);
+
+    assertThat(getFileMeasureAsInt(SECOND_PROJECT_FIRST_CLASS_FILE, "statements")).isEqualTo(3);
+  }
+
+  /* Complexity */
+
+  @Test
+  public void complexityAtProjectLevel() {
+    assertThat(getProjectMeasureAsInt("complexity")).isEqualTo(17);
+  }
+
+  @Test
+  public void complexityAtDirectoryLevel() {
+    assertThat(getDirectoryMeasureAsInt(FIRST_PROJECT_DIRECTORY, "complexity")).isEqualTo(14);
+    assertThat(getDirectoryMeasureAsInt(SECOND_PROJECT_DIRECTORY, "complexity")).isEqualTo(3);
+  }
+
+  @Test
+  public void complexityAtFileLevel() {
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_FIRST_CLASS_FILE, "complexity")).isEqualTo(13);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_SECOND_CLASS_FILE, "complexity")).isEqualTo(1);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_ASSEMBLY_INFO_FILE, "complexity")).isEqualTo(0);
+
+    assertThat(getFileMeasureAsInt(SECOND_PROJECT_FIRST_CLASS_FILE, "complexity")).isEqualTo(3);
+  }
+
+  /* Cognitive Complexity */
+
+  @Test
+  public void cognitiveComplexityAtDirectoryLevel() {
+    assertThat(getDirectoryMeasureAsInt(FIRST_PROJECT_DIRECTORY, "cognitive_complexity")).isEqualTo(2);
+    assertThat(getDirectoryMeasureAsInt(SECOND_PROJECT_DIRECTORY, "cognitive_complexity")).isEqualTo(2);
+  }
+
+  @Test
+  public void cognitiveComplexityAtProjectLevel() {
+    assertThat(getProjectMeasureAsInt("cognitive_complexity")).isEqualTo(4);
+  }
+
+  @Test
+  public void cognitiveComplexityAtFileLevel() {
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_FIRST_CLASS_FILE, "cognitive_complexity")).isEqualTo(2);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_SECOND_CLASS_FILE, "cognitive_complexity")).isEqualTo(0);
+    assertThat(getFileMeasureAsInt(FIRST_PROJECT_ASSEMBLY_INFO_FILE, "cognitive_complexity")).isEqualTo(0);
+
+    assertThat(getFileMeasureAsInt(SECOND_PROJECT_FIRST_CLASS_FILE, "cognitive_complexity")).isEqualTo(2);
+  }
```

Comment:
This makes me think that we need to add tasks for updating the metrics when will add support for C#9. One example would be to count the Record types (at least as classes if not as a separate construct).

Btw, do we count structs or enums?

Reply:
I took a look and we do consider structs and interfaces classes from the metrics point of view: https://github.com/SonarSource/sonar-dotnet/blob/734df7968037387faade9353c7aedd1a77c010a0/sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Common/MetricsTest.cs#L216.

Enums are **not** counted.

Reply:
thanks!